src/stores/MessageList.tsx

```diff
@@ -0,0 +1,160 @@
+import React, { useState } from 'react';
+import { observer } from 'mobx-react-lite';
+import { useStore } from '../hooks/useStore';
+import '../styles/message-list.scss';
+
+const Messages = () => {
+	const {
+		parsedMessagesHistory, clearParsedMessages, sendMessage, setFromFile, indicators, editMessageMode,
+		editedMessageIndex,
+	} = useStore();
+	const { addParsedMessage } = useStore();
+	const [loadedFile, setFile] = useState<string>('');
+
+	const fileLoader = (file: FileList | null) => {
+		if (file !== null) {
+			const reader = new FileReader();
+			reader.readAsText(file.item(0) as Blob);
+			reader.onload = () => {
+				jsonMessagesFromString(reader.result);
+			};
+		}
+	};
+
+	const jsonMessagesFromString = (rawFromFile: string | ArrayBuffer | null) => {
+		const json = JSON.parse(rawFromFile as string);
+		clearParsedMessages();
+
+		for (let i = 0; i < json.length; i++) {
+			addParsedMessage(JSON.stringify(json[i]));
+		}
+	};
+
+	const replay = (array: string[]) => {
+		setFromFile(true);
+		clearParsedMessages();
+		let longestDelay = 0;
+		array.forEach((item, index) => {
+			if (JSON.parse(item).delay && JSON.parse(item).delay >= longestDelay) {
+				longestDelay = index;
+			}
+		});
+
+		for (let i = 0; i < array.length; i++) {
+			if (JSON.parse(array[i]).delay) {
+				// eslint-disable-next-line no-loop-func
+				setTimeout(() => {
+					sendMessage(array[i] as unknown as object);
+					if (i === longestDelay) {
+						setFromFile(false);
+					}
+				},
+				JSON.parse(array[i]).delay);
+			} else {
+				sendMessage(array[i] as unknown as object);
+			}
+		}
+	};
+
+	return (
+		<div>
+			<div>
+			</div>
+			<div style={{ height: '430px', overflowY: 'scroll' }}>
+			 <MessageList items={parsedMessagesHistory} indicators={indicators} editMessageMode={editMessageMode}
+					editedMessageIndex={editedMessageIndex}/>
+			</div>
+			<div>
+				<button className="mainButton" onClick={() => { clearParsedMessages(); }}>
+                        Clear
+				</button>
+				<button className="mainButton" >
+					<a
+						href={`data:application/json;base64,${btoa(`[${parsedMessagesHistory.toString()}]`)}`} download>
```

Comment:
``data:application/json;base64,${btoa(`[${parsedMessagesHistory.toString()}]`)}``

This will be calculated every time component rerenders which might be expensive and unnecessary.
Let's instead use helper function to download files. You can use downloadFile fn from th2-infra-editorv2

Also let's make this button disabled if list is empty